veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,185 @@
+# VEP #20: Seamless TCP migration and vhost-user support with passt
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [x] (R) Enhancement issue created, linking to VEP directory in [kubevirt/enhancements] (not the initial VEP PR)
+
+## Overview
+
+This proposal outlines the enhancements implemented in passt and how KubeVirt will integrate with passt to benefit from them:
+- vhost-user protocol
+- Seamless live migration of TCP sessions
+
+## Motivation
+
+Currently, KubeVirt does not support all networking requirements expected by users:
+- Seamless network connectivity on migration
+- Observability/Service Mesh support
+- High throughput (on par with tap)
+- Support for all IP network protocols
+
+Bridge binding does not support observability or service meshes, Masquerade hides the real IP from the user, and passt currently lacks seamless migration and tap-level performance.
+
+**Most** of these gaps have been addressed in the recently released version of passt. The exception is support for non-TCP/UDP/ICMP protocols, which has not yet been implemented but is planned for future releases.
+
+To leverage these features, KubeVirt requires adaptation.
+
+## Goals
+
+- Seamless live migration of TCP connections without disconnection using passt network binding.
+- High network throughput with passt network binding.
+
+## Non Goals
+
+- Supporting network protocols other than TCP/UDP/ICMP (ARP/DNS).
+- ADressing the design passt itself.
+- Supporting CNIs that do not provide sticky IPs during migration (migration succeeds, but existing connections break).
+
+## Definition of Users
+
+- Cluster Admins
+- VM/Namespace owners
+- Guest VM users – typically unaware of their network binding
+
+## User Stories
+
+- As a Guest VM user, I want optimal network performance comparable to bridge and masquerade, ensuring my virtualized applications perform well.  
+- As a Cluster Admin, I want observability and service meshes for enhanced monitoring, encryption, and load balancing.
+- As a Cluster Admin, I require seamless VM migration to perform infrastructure upgrades without affecting business continuity.
+
+## Repos
+
+- [https://github.com/kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)
+
+## Design
+
+### Set TCP_REPAIR Socket Option
+
+During migration, active TCP sessions must transfer from one passt instance to another on a different host. The new instance cannot simply "hijack" the connection even if it knows all necessary details (IPs, ports, sequences). A privileged operation using the `TCP_REPAIR` socket option must be performed on both source and target sockets.
+
+Since passt runs in the unprivileged Linux context of virt-launcher, it lacks the necessary `CAP_NET_ADMIN` capability. Therefore, passt delegates this operation to a privileged helper running in virt-handler.
+
+#### The Role of passt-repair
```

Comment:
Once the high level solution is clear, please provide a diagram that shows the components in the virt-launcher and the interaction with the virt-handler.

In general, I think you should provide a description from top-to-down. I.e. first explain the big picture and then get into details. E.g. I would expect to see the flow first and only then details about each compoinent role.

Reply:
Thanks, looks good now. This can be resolved.